CONTRIBUTING.md

```diff
@@ -0,0 +1,217 @@
+
+## How to contribute to CRIU
+
+CRIU project is (almost) the never-ending story, because we have to always keep up with the
+Linux kernel supporting checkpoint and restore for all the features it provides. Thus we're
+looking for contributors of all kinds -- feedback, bug reports, testing, coding, writing, etc.
+Here are some useful hints to get involved.
+
+* We have both -- [very simple](https://github.com/checkpoint-restore/criu/issues?q=is%3Aissue+is%3Aopen+label%3Aenhancement) and [more sophisticated](https://github.com/checkpoint-restore/criu/issues?q=is%3Aissue+is%3Aopen+label%3A%22new+feature%22) coding tasks;
+* CRIU does need [extensive testing](https://github.com/checkpoint-restore/criu/issues?q=is%3Aissue+is%3Aopen+label%3Atesting);
+* Documentation is always hard, we have [some information](https://criu.org/Category:Empty_articles) that is to be extracted from people's heads into wiki pages as well as [some texts](https://criu.org/Category:Editor_help_needed) that all need to be converted into useful articles;
+* Feedback is expected on the github issues page and on the [mailing list](https://lists.openvz.org/mailman/listinfo/criu);
+* We accept github pull requests and this is the preferred way to contribute to CRIU. If you prefer to send patches by email, you are welcome to send them to [CRIU development mailing list](https://lists.openvz.org/mailman/listinfo/criu).
+Below we describe in more detail recommend practices for CRIU developemnt.
+* Spread the word about CRIU in [social networks](http://criu.org/Contacts);
+* If you're giving a talk about CRIU -- let us know, we'll mention it on the [wiki main page](https://criu.org/News/events);
+
+### Seting up the developemnt environment
+
+Although criu could be run as non-root (see [Security](https://criu.org/Security), development is better to be done as root. For example, some tests require root. So, it would be a good idea to set up some recent Linux distro on a virtual machine.
+
+### Get the source code
+
+The CRIU sources are tracked by Git. Official CRIU repo is at https://github.com/checkpoint-restore/criu.
+
+The repository may contain multiple branches. Development happens in the **criu-dev** branch.
+
+To clone CRIU repo and switch to the proper branch, run:
+
+```
+        git clone https://github.com/checkpoint-restore/criu criu
+        cd criu
+        git checkout criu-dev
+```
+
+### Compile
+
+First, you need to install compile-time dependencies. Check [Installation dependencies](https://criu.org/Installation#Dependencies) for more info.
+
+To compile CRIU, run:
+
+```
+        make
+```
+
+This should create the `./criu/criu` executable.
+
+## Edit the source code
+
+If you use ctags, you can generate the ctags file by running
+
+```
+        make tags
+```
+
+When you change the source code, please keep in mind the following code conventions:
+
+* we prefer tabs and indentations to be 8 characters width
```

Comment:
except for python files where we use 4 spaces instead of tabs (e.g. https://github.com/checkpoint-restore/criu/commit/5aa72e723707e2bd7e8ed9841c2ad392781d066d)

Reply:
@rst0git I agree in general, but as I as l said, let's start from importing this from wiki to github and then update it.